### PR TITLE
Undertaking #54, generating sln, vcxproj, vcxproj.filers and packages.config

### LINKED ISSUE #54: Generate visual studio builds.
`libbitcoin-explorer` already does this, so that implementation will need to be removed in favor of the file-search technique employed by `libbitcoin-build`

### PR DESCRIPTION
**Pull request by pmienk**
Provides generation of VS2013 artifacts aiming to resolve #54.  Intending further generalization for greater support.

### CODE REVIEW

**Comment by pmienk** [comment]
Provides generation of VS2013 artifacts aiming to resolve #54.  Intending further generalization for greater support.

File: templates/gsl.vcxproj [hunk 1]
```diff
@@ -241,6 +241,14 @@ function is_product_library(product)
     return (my.product.prefix = "lib")
 endfunction
 
+function is_compiler_ctp(product)
+    define my.product = is_compiler_ctp.product
+    return is_product_library(my.product) |\
+        (is_product_binary(my.product) & !is_option_tests(my.product)) |\
+        (is_product_binary(my.product) & is_option_tests(my.product) &\
```
Comment (evoskuil) [inline templates/gsl.vcxproj]: I can't see a reason for a libconsensus exception here. It should halve been the same as the other libs.

Comment (pmienk) [reply templates/gsl.vcxproj]: A removal of the condition and recompilation of libbitcoin-consensus results in the following error(s):
```
z:\git\libbitcoin-consensus\src\clone\serialize.h(40): error C2146: syntax error : missing ';' before identifier 'deser
ialize_type' (..\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\builds\msvc\vs
2013\libbitcoin-consensus-test\libbitcoin-consensus-test.vcxproj]
z:\git\libbitcoin-consensus\src\clone\serialize.h(40): error C4430: missing type specifier - int assumed. Note: C++ doe
s not support default-int (..\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\b
uilds\msvc\vs2013\libbitcoin-consensus-test\libbitcoin-consensus-test.vcxproj]
z:\git\libbitcoin-consensus\src\clone\serialize.h(814): error C2144: syntax error : 'bool' should be preceded by ';' (.
.\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\builds\msvc\vs2013\libbitcoin
-consensus-test\libbitcoin-consensus-test.vcxproj]
z:\git\libbitcoin-consensus\src\clone\serialize.h(814): error C4430: missing type specifier - int assumed. Note: C++ do
es not support default-int (..\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\
builds\msvc\vs2013\libbitcoin-consensus-test\libbitcoin-consensus-test.vcxproj]
z:\git\libbitcoin-consensus\src\clone\serialize.h(818): error C2144: syntax error : 'bool' should be preceded by ';' (.
.\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\builds\msvc\vs2013\libbitcoin
-consensus-test\libbitcoin-consensus-test.vcxproj]
z:\git\libbitcoin-consensus\src\clone\serialize.h(818): error C4430: missing type specifier - int assumed. Note: C++ do
es not support default-int (..\..\..\..\test\consensus__verify_flags_to_script_flags.cpp) [z:\git\libbitcoin-consensus\
builds\msvc\vs2013\libbitcoin-consensus-test\libbitcoin-consensus-test.vcxproj]
msbuild /verbosity:minimal /p:Platform=x64 /p:Configuration=StaticRelease libbitcoin-consensus.sln failed.
Building libbitcoin-consensus failed.
```

Comment (evoskuil) [reply templates/gsl.vcxproj]: Looks like it wasn't an oversight, just forgot about this issue.

**Comment by evoskuil** [comment]
Apart from version numbers the relevant aspects of solution/project/filter schema haven't changed across 2010/12/13/15/17. As such might it make sense to generate from a common VS template so that we don't have as much redundant template code?

File: generate.xml [hunk 2]
```diff
@@ -1906,24 +1906,22 @@
...
+      <package label="zmq"             name="libzmq_vc120"                    version="4.2.30" />
     </nuget>
-    <!--
     <nuget build_name="zmq" compiler="vc140">
-      <package label="zmq"             name="libzmq_vc140"                    version="4.2.20.1" />
+      <package label="zmq"             name="libzmq_vc140"                    version="4.2.30" />
     </nuget>
     <nuget build_name="zmq" compiler="vc141">
-      <package label="zmq"             name="libzmq-vc141"                    version="4.2.2" />
+      <package label="zmq"             name="libzmq-vc141"                    version="4.2.30" />
```
Comment (evoskuil) [inline generate.xml]: This is the wrong package, https://www.nuget.org/packages/libzmq-vc141/ vs https://www.nuget.org/packages/libzmq_vc141/

File: generate.xml [hunk 3]
```diff
@@ -1906,24 +1906,22 @@
       <package label="test"            name="boost_unit_test_framework-vc141" version="1.64.0.0" />
     </nuget>
     <nuget build_name="secp256k1" compiler="vc120" >
-      <package label="secp256k1"       name="secp256k1_vc120"                 version="0.1.0.14" />
+      <package label="secp256k1"       name="secp256k1_vc120"                 version="0.1.0.16" />
     </nuget>
     <nuget build_name="secp256k1" compiler="vc140" >
-      <package label="secp256k1"       name="secp256k1_vc140"                 version="0.1.0.14" />
+      <package label="secp256k1"       name="secp256k1_vc140"                 version="0.1.0.15" />
```
Comment (evoskuil) [inline generate.xml]: This is the correct package, but from another owner: https://www.nuget.org/packages/secp256k1_vc140/ I've asked him if he will remove so I can continue to publish the full series.

**Pull request closed without merging**